oshmem/mca/memheap/base/memheap_base_static.c

```diff
@@ -138,6 +140,26 @@ static int _check_perms(const char *perms)
     return OSHMEM_ERROR;
 }
 
+static int _check_non_static_segment(const map_segment_t *mem_segs,
+                                     int n_segment,
+                                     const void *start, const void *end)
+{
+    int i;
+
+    for (i = 0; i < n_segment; i++) {
+        if ((start <= mem_segs[i].super.va_base) &&
+            (mem_segs[i].super.va_base < end)) {
+            MEMHEAP_VERBOSE(100,
+                            "non static segment: %p-%p already exists as %p-%p",
+                            start, end, mem_segs[i].super.va_base,
+                            mem_segs[i].super.va_end);
+            return OSHMEM_ERROR;
+        }
+    }
+
+    return OSHMEM_SUCCESS;
+}
+
```

Comment:
When looking for existing static segments, we need to skip the non static which has previously been created, else it is added twice to the list which is an issue, although I don't have a reproducer now. 

In previous patch we would try to make sure to always be above `_end`, and there is code to skip when > `_end`.

As we look for known existing segments, it can only be contained inside the one we iterate for, I don't see a case where there can be partial overlap.